### PR TITLE
Add symbolic constants for vm opcode flags

### LINKED ISSUE #1057: Add symbolic constants for performing increment in vm.cpp
This would make them more readable. See #1039

### PR DESCRIPTION
**Pull request by zherczeg**
Should fix #1057.

### CODE REVIEW

File: jerry-core/vm/vm.h [hunk 1]
```diff
@@ -29,12 +29,25 @@
  * @{
  */
 
+/**
+ * Shift for extracting the type of getting input data.
```
Comment (akiss77) [inline jerry-core/vm/vm.h]: The description of these four macros is a bit hard to grasp. Would be nice to have them improved. Otherwise, the newly introduced macros look good.

Comment (LaszloLango) [inline jerry-core/vm/vm.h]: I think we usually write "position" instead of "shift".

**Comment by LaszloLango** [comment]
LGTM

**Comment by zherczeg** [comment]
Reworked the patch. Binary is reduced by 220 bytes.

File: jerry-core/vm/vm.h [hunk 2]
```diff
@@ -29,33 +29,71 @@
...
-#define VM_OC_GET_DATA_SHIFT 24
-#define VM_OC_GET_DATA_MASK 0x1f
-#define VM_OC_GET_DATA_CREATE_ID(V) \
-  (((V) & VM_OC_GET_DATA_MASK) << VM_OC_GET_DATA_SHIFT)
-#define VM_OC_GET_DATA_GET_ID(O) \
-  (((O) >> VM_OC_GET_DATA_SHIFT) & VM_OC_GET_DATA_MASK)
+/**
+ * Each CBC opcode is transformed to three vm opcodes:
+ *
+ *  - first opcode is a "get arguments" opcode which specify
```
Comment (akiss77) [inline jerry-core/vm/vm.h]: typo: specify -> specifies (three occurences)

File: jerry-core/vm/vm.c [hunk 3]
```diff
@@ -153,15 +153,15 @@ vm_init (ecma_compiled_code_t *program_p) /**< pointer to byte-code data */
 /**
  * Decode table for opcodes.
  */
-static const uint32_t vm_decode_table[] =
+static const uint16_t vm_decode_table[] =
```
Comment (akiss77) [inline jerry-core/vm/vm.c]: No blocker but nasty: this change does not seem to be related to the patch at all. I'd love to see separate ideas in separate patches/PRs.

Comment (zherczeg) [inline jerry-core/vm/vm.c]: I rework the commit log then.

**Comment by zherczeg** [comment]
Patch refocused a bit, and the description is aligned for the new focus.

File: jerry-core/vm/vm.h (hunk 2 above)
Comment (akiss77) [inline jerry-core/vm/vm.h]: was this comment ignored intentionally?

Comment (zherczeg) [inline jerry-core/vm/vm.h]: Forgot to fix it.

**Comment by LaszloLango** [comment]
Still LGTM

**Comment by akiss77** [comment]
LGTM
